### ISSUE TITLE
Support HomeKit Secure Video

### ISSUE THREAD

**Issue by ouvoun** [potmat/homebridge-google-nest-sdm#11]
First of all, thanks so much for your effort on this project. If you open up sponsorships I'm sure there are some of us who would be happy to chip in to buy you some drinks to show our gratitude. 😄 

I'm curious if this plugin could feasibly support HKSV, now that it's been added to homebridge? I don't know if there's any technical limitations that could make that difficult, but I wanted to open an issue for it just to discuss the effort involved. If you have no plans for it no worries, feel free to close.

**Comment by potmat** [comment]
Hi @ouvoun. Thank you so much for your support, I'm not sure what you mean by sponsorships.  There's a PayPal donate button in the readme.  Is there some other sponsorship scheme I'm not aware of?

I see a few issues with this:

1. The battery powered cameras will only stream for five minutes.  At any rate continuous streaming from them would likely wear the battery down quickly, so it's probably not a good idea.
2. Continuous streaming from the wired cameras is possible in theory, but in practice you have to make a request to "extend" the stream every five minutes.  I've not been able to figure out how to actually do this yet. See this issue: https://stackoverflow.com/questions/68433468/how-to-properly-use-the-extendrtspstream-command-in-googles-nest-device-acces
3. If we instead only record a clip when an event occurs the clip will likely lag 10 or more seconds behind the actual event, which may not be useful.  This is due to the fact that setting up the stream and starting recording takes time.
4. The API is rate limited, if there is a lot of events occurring Google will likely reject some stream requests.

That said, I haven't even looked into HKSV yet, I was more focused on just being able to view the cameras.  I'll take a look when I get some time.

**Comment by potmat** [comment]
@evk0122 : I agree that recording clips on demand is probably the only way this will work.  I don't see how it would be a "jail break" though, provided you've paid your $5 and are using it for personal non-commercial use I can't see what rule you'd be breaking by recording event clips.  I reviewed the terms and conditions and can't find anything that would indicate such usage would be a violation (obligatory "I am not a lawyer").

If anyone out there can determine that such usage would be a violation of the TOS please let us know!

**Comment by ouvoun** [comment]
Thanks for the context. Also I must’ve missed the donation button since it was so small; sent a donation your way!

Personally, I feel like event-triggered recordings would still be quite nice, since it’d let the event triggers be defined within the Nest world. The delay is definitely a drawback, but as @evk0122 mentioned the Google app is already so delayed that it’s already useless for live events. 😆 

Does the API provide a trigger for _ending_ the event as well, or just starting? Within the Google app at least it seems like motion-triggered recordings will continue past the standard duration if movement continues or another trigger occurs.

**Comment by potmat** [comment]
@ouvoun : Thank you so much!

Thanks for your input, it seems like event recordings are the way to go.

Unfortunately there is no "event ended" trigger. So we would have to decide on some standard duration for event recordings like 10 seconds.

**Comment by potmat** [comment]
Just realizing that I don't own a HomePod, AppleTV, or iPad.  Correct me if I'm wrong but you need one of those to have HKSV right?

**Comment by ouvoun** [comment]
Unfortunately, yes.

**Comment by potmat** [comment]
Perhaps I should enable the GitHub "sponsor" button 😄.

I'll try to find a HomePod mini on eBay or something.  I bought a second generation cam out of pocket to get that working, even though I have no use for it (it's just sitting on my desk pointing at the wall) and I'm not super keen to do that again.

**Comment by ouvoun** [comment]
@potmat Sent another donation your way, maybe some other folks can chip in as well so you're able to pick up a HomePod. :)

**Comment by potmat** [comment]
@ouvoun: Thank you so much!  I managed to find a HomePod used nearby.  I've started on HKSV but haven't got it working yet.  Since it was only just added to Homebridge there isn't much reference material for it yet.

**Comment by potmat** [comment]
Just keeping everyone up to date, I have actually gotten HKSV working here in my house.  However, there are a few things maybe we should discuss.

Because I'm not doing continuous recording (for the reasons mentioned above) HKSV doesn't work exactly as you'd expect.  If you simply walk through a room, you are not likely to see an event.  This is because by the time the stream is setup and Homebridge is actually sending video to your hub(s) you've already left the room, and when your hub analyzes the video it won't see any motion, so no event will be registered.

So far I've only made it work with software encoding, which means that you need pretty serious hardware.  My Pi4 will quickly reach 100% CPU even with only a single camera recording events, if there are multiple cameras recording you can use it as a small space heater.

I can still release even with these limitations and claim that HKSV support is "beta" or something.  If anyone has any thoughts please chime in...

**Comment by ouvoun** [comment]
Interesting. So the mere presence of the stream isn't enough for HKSV to start recording...it also needs to do its own motion analysis?

Either way, love the idea of pushing it out as a beta. Thanks for your work on this!

**Comment by evk0122** [comment]
I was thinking to install homebridge on my antique Mac mini and try to make dusty c2d do some weightlifting. 
Maybe we can figure out what else can be tried once we have the beta. I never used a device with HKSV support and since it was just added to Homebridge I also didn't have time to see other examples. 
In case of doorbell the event can be triggered not only by walking but button press. It actually makes even more sense - the motion detection should be triggered before the doorbell is pressed. Therefore by the time someone actually presses the button - the stream should be half-way through.

**Comment by potmat** [comment]
Hey all, just published 1.1.0-beta.1 with HKSV support.

You can install it manually with

`npm install homebridge-google-nest-sdm@beta`

Let me know...

**Comment by potmat** [comment]
Just noticed my cameras are no longer sending any events, motion, person, even doorbell ring.  Perhaps Google cut me off? Anyone else experiencing this?

**Comment by ouvoun** [comment]
I just gave it a spin.

I do see the expected HKSV settings within my wired camera (yay), but interestingly enough it seems no clips are actually recorded, even when receiving motion events from the Google app. I walked around for a few minutes to make sure it wasn't just the delay issue you mentioned above, but it seems to be different.

Oddly enough I've never properly received notifications for this camera — only for my doorbell. That continues to be the case, it seems.

**Comment by potmat** [comment]
@ouvoun : If you're using the second-gen cameras I've noticed that they report far fewer events than the first gen.  For example: The first gen cameras will report a motion/person event every time you get up off the couch, the second gen cameras will report a person event when you walk in the room, but as you move around in the room they will not.

Or it could be that you were experiencing this issue: https://stackoverflow.com/questions/71417968/cameras-no-longer-reporting-events

**Comment by ouvoun** [comment]
It definitely could be the delay, but I did try walking around for at least a few minutes. I'd expect that to be well above the threshold?

It very well might be a Google issue, but considering it's still not working as of now I'm not sure that it's related.

I should try clearing the device cache, maybe?

**Comment by potmat** [comment]
@ouvoun : Must be something with the second gen cameras, my first gen cameras are working perfectly.  But you're right, I've tried everything to make my second gen camera report an event and so far nothing....

Unfortunately if Google does not raise any events there's nothing the plugin can do.

**Comment by potmat** [comment]
Anyone else still playing with this. As it seems to sort of work with the 1st generation cameras I have no problem releasing it.

**Comment by krehman** [comment]
@potmat I installed the beta and got everything setup (I think), but still not able to get any HKSV recordings. Using Nest Hello (wired) with Apple TV.

**Comment by potmat** [comment]
@krehman: The wired doorbell should work (mine does).  Can you send me any log messages (you may need to turn on debug mode) around the time of a motion event?

**Comment by krehman** [comment]
@potmat See below. I tried to remove the token / encrypted info. Let me know if you need anything else. 

HKSV does trigger motion events if enable notifications, but it just isn't recording it.

Thanks

```
[4/8/2022, 5:39:11 PM] [homebridge-google-nest-sdm] Recording active set to false
[4/8/2022, 5:39:11 PM] [homebridge-google-nest-sdm] Recording active set to true
[4/8/2022, 5:39:11 PM] [homebridge-google-nest-sdm] Recording active set to true
[4/8/2022, 5:40:02 PM] [homebridge-google-nest-sdm] Event received: {
  "eventId": "02bbd6fa-28ad-4dcb-bbc9-e4b964ea96e8",
  "timestamp": "2022-04-08T21:39:58.865Z",
  "resourceUpdate": {
    "name": "enterprises/491ef7bf-60cc-4fb4-b99b-d724265bfcac/devices/AVPHwEvNnkwQb_k0_eaOA",
    "events": {
      "sdm.devices.events.CameraMotion.Motion": {
        "eventSessionId": "AVPHwEvITT3sE8j98d1qSuZn9YUbA",
        "eventId": "CiUA2vuxr0605jXyeRWdjGLUy--OAtNpJ6G5bjwUsa5CULPm14wFbp6u_tOr7S6A"
      }
    }
  },
  "userId": "AVPHwEvzcQflv",
  "resourceGroup": ["enterprises/491ef7bf-60cc-4fb4-b99b-d724265bfcac/devices/AVPHwEvNnkwQb_T7AL1IUoxr4OUa_x]3Rk0_eaOA"]
}
[4/8/2022, 5:40:02 PM] [homebridge-google-nest-sdm] Executing command sdm.devices.commands.CameraEventImage.GenerateImage with parameters {"eventId":"CiUA2vuxr0605jXyeRWdjGLUy"} Front Door Doorbell
[4/8/2022, 5:40:03 PM] [homebridge-google-nest-sdm] Execution of command sdm.devices.commands.CameraEventImage.GenerateImage returned {"url":"https://nexusapi-us1.dropcam.com/sdm_resource/mnk/snapshot","token":"g.0..."} Front Door Doorbell
[4/8/2022, 5:40:03 PM] [homebridge-google-nest-sdm] Motion detected! Front Door Doorbell
[4/8/2022, 5:40:03 PM] [homebridge-google-nest-sdm] Recording request received.
[4/8/2022, 5:40:03 PM] [homebridge-google-nest-sdm] Executing command sdm.devices.commands.CameraLiveStream.GenerateRtspStream with parameters undefined Front Door Doorbell
[4/8/2022, 5:40:04 PM] [homebridge-google-nest-sdm] Execution of command sdm.devices.commands.CameraLiveStream.GenerateRtspStream returned {"streamUrls":{"rtspUrl":"rtsps://stream-us1-bravo.dropcam.com:443/sdm_live_stream/IS?auth=g.0."},"streamExtensionToken":"--euphM","streamToken":"g.0...","expiresAt":"2022-04-08T21:45:04.060Z"} Front Door Doorbell
[4/8/2022, 5:40:04 PM] [homebridge-google-nest-sdm] HksvStreamer start command received.
[4/8/2022, 5:40:04 PM] [homebridge-google-nest-sdm] ffmpeg -fflags +discardcorrupt+nobuffer -i rtsps://stream-us1-bravo.dropcam.com:443/ -acodec libfdk_aac -profile:a aac_eld -ar 48k -b:a 64k -ac 1 -f mp4 -an -sn -dn -codec:v libx264 -pix_fmt yuv420p -profile:v high -level:v 4.0 -b:v 2000k -force_key_frames expr:eq(t,n_forced*4) -r 30 -fflags +genpts -reset_timestamps 1 -movflags frag_keyframe+empty_moov+default_base_moof tcp://127.0.0.1:43897
[4/8/2022, 5:40:04 PM] [homebridge-google-nest-sdm] ffmpeg version 5.0 Copyright (c) 2000-2022 the FFmpeg developers
  built with gcc 9.3.0 (Alpine 9.3.0)
  configuration: --pkgconfigdir=/build/workspace/lib/pkgconfig --prefix=/build/workspace --pkg-config-flags=--static --extra-cflags=-I/build/workspace/include --extra-ldflags=-L/build/workspace/lib --extra-libs='-lpthread -lm' --enable-static --disable-debug --disable-shared --disable-ffplay --disable-doc --enable-openssl --enable-gpl --enable-version3 --enable-nonfree --enable-pthreads --enable-libvpx --enable-libmp3lame --enable-libopus --enable-libtheora --enable-libvorbis --enable-libx264 --enable-runtime-cpudetect --enable-libfdk-aac --enable-avfilter --enable-libopencore_amrwb --enable-libopencore_amrnb --enable-filters --enable-decoder=h264 --enable-network --enable-protocol=tcp --enable-libspeex --enable-demuxer=rtsp --enable-indev=alsa --enable-outdev=alsa

[4/8/2022, 5:40:04 PM] [homebridge-google-nest-sdm]   libavutil      57. 17.100 / 57. 17.100
  libavcodec     59. 18.100 / 59. 18.100
  libavformat    59. 16.100 / 59. 16.100
  libavdevice    59.  4.100 / 59.  4.100
  libavfilter     8. 24.100 /  8. 24.100
  libswscale      6.  4.100 /  6.  4.100
  libswresample   4.  3.100 /  4.  3.100
  libpostproc    56.  3.100 / 56.  3.100

[4/8/2022, 5:40:05 PM] [homebridge-google-nest-sdm] Recording active set to false
[4/8/2022, 5:40:05 PM] [homebridge-google-nest-sdm] Recording active set to true
[4/8/2022, 5:40:05 PM] [homebridge-google-nest-sdm] Recording active set to true
[4/8/2022, 5:40:05 PM] [homebridge-google-nest-sdm] Recording active set to false
[4/8/2022, 5:40:06 PM] [homebridge-google-nest-sdm] Input #0, rtsp, from 'rtsps://stream-us1-bravo.dropcam.com:443/sdm_live_stream/--?auth=g.0...EImTaoBT4gXSoddwuRHJH2----XvpY-':
  Metadata:
    title           : SDM
  Duration: N/A, start: -0.347000, bitrate: N/A
[4/8/2022, 5:40:06 PM] [homebridge-google-nest-sdm] 
  Stream #0:0: Audio: aac (LC), 48000 Hz, stereo, fltp
  Stream #0:1: Video: h264 (Main), yuvj420p(pc, bt709, progressive), 1600x1200 [SAR 1:1 DAR 4:3], 30 fps, 30 tbr, 90k tbn

[4/8/2022, 5:40:06 PM] [homebridge-google-nest-sdm] Stream mapping:
  Stream #0:1 -> #0:0 (h264 (native) -> h264 (libx264))
Press [q] to stop, [?] for help

[4/8/2022, 5:40:09 PM] [homebridge-google-nest-sdm] [swscaler @ 0x555cb9c53e00] [swscaler @ 0x555cb9c60e00] deprecated pixel format used, make sure you did set range correctly
[4/8/2022, 5:40:09 PM] [homebridge-google-nest-sdm] [swscaler @ 0x555cb9c53e00] [swscaler @ 0x555cb97aaf80] deprecated pixel format used, make sure you did set range correctly
[4/8/2022, 5:40:09 PM] [homebridge-google-nest-sdm] [swscaler @ 0x555cb9c53e00] [swscaler @ 0x555cb9d09000] deprecated pixel format used, make sure you did set range correctly
[4/8/2022, 5:40:09 PM] [homebridge-google-nest-sdm] [swscaler @ 0x555cb9c53e00] [swscaler @ 0x555cb9d46580] deprecated pixel format used, make sure you did set range correctly
[4/8/2022, 5:40:09 PM] [homebridge-google-nest-sdm] [swscaler @ 0x555cb9c53e00] [swscaler @ 0x555cb9d85480] deprecated pixel format used, make sure you did set range correctly
[4/8/2022, 5:40:09 PM] [homebridge-google-nest-sdm] [libx264 @ 0x555cb9769a80] using SAR=1/1
[4/8/2022, 5:40:09 PM] [homebridge-google-nest-sdm] [libx264 @ 0x555cb9769a80] using cpu capabilities: MMX2 SSE2Fast SSSE3 SSE4.2
[4/8/2022, 5:40:09 PM] [homebridge-google-nest-sdm] [libx264 @ 0x555cb9769a80] profile High, level 4.0, 4:2:0, 8-bit
[4/8/2022, 5:40:09 PM] [homebridge-google-nest-sdm] [libx264 @ 0x555cb9769a80] 264 - core 163 - H.264/MPEG-4 AVC codec - Copyleft 2003-2021 - http://www.videolan.org/x264.html - options: cabac=1 ref=3 deblock=1:0:0 analyse=0x3:0x113 me=hex subme=7 psy=1 psy_rd=1.00:0.00 mixed_ref=1 me_range=16 chroma_me=1 trellis=1 8x8dct=1 cqm=0 deadzone=21,11 fast_pskip=1 chroma_qp_offset=-2 threads=6 lookahead_threads=1 sliced_threads=0 nr=0 decimate=1 interlaced=0 bluray_compat=0 constrained_intra=0 bframes=3 b_pyramid=2 b_adapt=1 b_bias=0 direct=1 weightb=1 open_gop=0 weightp=2 keyint=250 keyint_min=25 scenecut=40 intra_refresh=0 rc_lookahead=40 rc=abr mbtree=1 bitrate=2000 ratetol=1.0 qcomp=0.60 qpmin=0 qpmax=69 qpstep=4 ip_ratio=1.40 aq=1:1.00
[4/8/2022, 5:40:09 PM] [homebridge-google-nest-sdm] Output #0, mp4, to 'tcp://127.0.0.1:43897':
  Metadata:
    title           : SDM
    encoder         : Lavf59.16.100
  Stream #0:0: Video: h264 (avc1 / 0x31637661), yuv420p(tv, bt709, progressive), 1600x1200 [SAR 1:1 DAR 4:3], q=2-31, 2000 kb/s, 30 fps, 15360 tbn
    Metadata:
      encoder         : Lavc59.18.100 libx264
    Side data:
      cpb: bitrate max/min/avg: 0/0/2000000 buffer size: 0 vbv_delay: N/A

[4/8/2022, 5:40:09 PM] [homebridge-google-nest-sdm] mp4 box type ftyp and length 20
[4/8/2022, 5:40:09 PM] [homebridge-google-nest-sdm] mp4 box type moov and length 812
[4/8/2022, 5:40:14 PM] [homebridge-google-nest-sdm] frame=  121 fps= 37 q=23.0 size=       1kB time=00:00:02.26 bitrate=   3.0kbits/s dup=120 drop=0 speed=0.696x    
[4/8/2022, 5:40:14 PM] [homebridge-google-nest-sdm] frame=  122 fps= 15 q=22.0 size=       1kB time=00:00:02.30 bitrate=   2.9kbits/s dup=120 drop=0 speed=0.274x    
[4/8/2022, 5:40:15 PM] [homebridge-google-nest-sdm] frame=  126 fps= 14 q=22.0 size=       1kB time=00:00:02.43 bitrate=   2.8kbits/s dup=120 drop=0 speed=0.272x    
[4/8/2022, 5:40:15 PM] [homebridge-google-nest-sdm] frame=  130 fps= 14 q=22.0 size=       1kB time=00:00:02.56 bitrate=   2.6kbits/s dup=120 drop=0 speed=0.27x    
[4/8/2022, 5:40:16 PM] [homebridge-google-nest-sdm] frame=  135 fps= 13 q=21.0 size=       1kB time=00:00:02.73 bitrate=   2.5kbits/s dup=120 drop=0 speed=0.269x    
[4/8/2022, 5:40:16 PM] [homebridge-google-nest-sdm] frame=  137 fps= 13 q=21.0 size=       1kB time=00:00:02.80 bitrate=   2.4kbits/s dup=120 drop=0 speed=0.26x    
[4/8/2022, 5:40:17 PM] [homebridge-google-nest-sdm] frame=  144 fps= 12 q=21.0 size=       1kB time=00:00:03.03 bitrate=   2.2kbits/s dup=120 drop=0 speed=0.262x    
[4/8/2022, 5:40:18 PM] [homebridge-google-nest-sdm] HksvStreamer destroy command received, ending process.
[4/8/2022, 5:40:18 PM] [homebridge-google-nest-sdm] Executing command sdm.devices.commands.CameraLiveStream.StopRtspStream with parameters {"streamExtensionToken":"-----------gpwMCEublSNwa-]----Qf6N-Bez74c-"} Front Door Doorbell
[4/8/2022, 5:40:18 PM] [homebridge-google-nest-sdm] Execution of command sdm.devices.commands.CameraLiveStream.StopRtspStream returned undefined Front Door Doorbell
[4/8/2022, 5:40:21 PM] [homebridge-google-nest-sdm] mp4 box type moof and length 1056
[4/8/2022, 5:40:23 PM] [homebridge-google-nest-sdm] frame=  146 fps=8.6 q=-1.0 Lsize=     986kB time=00:00:04.76 bitrate=1693.8kbits/s dup=120 drop=0 speed=0.279x    
video:983kB audio:0kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: 0.231099%

[4/8/2022, 5:40:23 PM] [homebridge-google-nest-sdm] [libx264 @ 0x555cb9769a80] frame I:2     Avg QP:13.35  size:268112
[libx264 @ 0x555cb9769a80] frame P:37    Avg QP:14.18  size: 10825
[libx264 @ 0x555cb9769a80] frame B:107   Avg QP:21.84  size:   650

[4/8/2022, 5:40:23 PM] [homebridge-google-nest-sdm] [libx264 @ 0x555cb9769a80] consecutive B-frames:  2.1%  0.0%  2.1% 95.9%
[libx264 @ 0x555cb9769a80] mb I  I16..4: 26.8% 21.5% 51.6%
[libx264 @ 0x555cb9769a80] mb P  I16..4:  0.0%  0.0%  0.0%  P16..4: 20.4%  0.9%  1.9%  0.0%  0.0%    skip:76.7%
[libx264 @ 0x555cb9769a80] mb B  I16..4:  0.0%  0.0%  0.0%  B16..8: 10.4%  0.0%  0.0%  direct: 0.0%  skip:89.5%  L0:12.1% L1:87.7% BI: 0.2%
[libx264 @ 0x555cb9769a80] final ratefactor: 17.09
[libx264 @ 0x555cb9769a80] 8x8 transform intra:21.7% inter:49.7%
[libx264 @ 0x555cb9769a80] coded y,uvDC,uvAC intra: 78.2% 61.0% 28.3% inter: 2.6% 1.2% 0.2%
[libx264 @ 0x555cb9769a80] i16 v,h,dc,p: 21% 28% 38% 13%
[libx264 @ 0x555cb9769a80] i8 v,h,dc,ddl,ddr,vr,hd,vl,hu: 21% 33% 23%  4%  3%  3%  5%  2%  5%
[libx264 @ 0x555cb9769a80] i4 v,h,dc,ddl,ddr,vr,hd,vl,hu: 27% 32% 14%  4%  4%  4%  5%  4%  6%
[libx264 @ 0x555cb9769a80] i8c dc,h,v,p: 50% 28% 19%  2%
[libx264 @ 0x555cb9769a80] Weighted P-Frames: Y:0.0% UV:0.0%
[libx264 @ 0x555cb9769a80] ref P L0: 86.5%  0.6% 11.2%  1.7%
[libx264 @ 0x555cb9769a80] ref B L0: 78.9% 20.1%  1.0%
[libx264 @ 0x555cb9769a80] ref B L1: 87.2% 12.8%
[libx264 @ 0x555cb9769a80] kb/s:1654.09

[4/8/2022, 5:40:23 PM] [homebridge-google-nest-sdm] Exiting normally, received signal 15.
[4/8/2022, 5:40:23 PM] [homebridge-google-nest-sdm] Socket destroyed.
[4/8/2022, 5:40:23 PM] [homebridge-google-nest-sdm] Recording request received.
[4/8/2022, 5:40:23 PM] [homebridge-google-nest-sdm] Executing command sdm.devices.commands.CameraLiveStream.GenerateRtspStream with parameters undefined Front Door Doorbell
[4/8/2022, 5:40:23 PM] [homebridge-google-nest-sdm] Execution of command sdm.devices.commands.CameraLiveStream.GenerateRtspStream returned {"streamUrls":{"rtspUrl":"rtsps://stream-us1-bravo.dropcam.com:443/sdm_live_stream/---uRq-3iQ5---O8---jG732vDR54N289FGP-","streamToken":"g.0...-","expiresAt":"2022-04-08T21:45:23.685Z"} Front Door Doorbell
[4/8/2022, 5:40:23 PM] [homebridge-google-nest-sdm] HksvStreamer start command received.
[4/8/2022, 5:40:23 PM] [homebridge-google-nest-sdm] ffmpeg -fflags +discardcorrupt+nobuffer -i rtsps://stream-us1-bravo.dropcam.com:443/sdm_live_stream/had -acodec libfdk_aac -profile:a aac_eld -ar 48k -b:a 64k -ac 1 -f mp4 -an -sn -dn -codec:v libx264 -pix_fmt yuv420p -profile:v high -level:v 4.0 -b:v 2000k -force_key_frames expr:eq(t,n_forced*4) -r 30 -fflags +genpts -reset_timestamps 1 -movflags frag_keyframe+empty_moov+default_base_moof tcp://127.0.0.1:46069
[4/8/2022, 5:40:23 PM] [homebridge-google-nest-sdm] ffmpeg version 5.0 Copyright (c) 2000-2022 the FFmpeg developers
  built with gcc 9.3.0 (Alpine 9.3.0)
  configuration: --pkgconfigdir=/build/workspace/lib/pkgconfig --prefix=/build/workspace --pkg-config-flags=--static --extra-cflags=-I/build/workspace/include --extra-ldflags=-L/build/workspace/lib --extra-libs='-lpthread -lm' --enable-static --disable-debug --disable-shared --disable-ffplay --disable-doc --enable-openssl --enable-gpl --enable-version3 --enable-nonfree --enable-pthreads --enable-libvpx --enable-libmp3lame --enable-libopus --enable-libtheora --enable-libvorbis --enable-libx264 --enable-runtime-cpudetect --enable-libfdk-aac --enable-avfilter --enable-libopencore_amrwb --enable-libopencore_amrnb --enable-filters --enable-decoder=h264 --enable-network --enable-protocol=tcp --enable-libspeex --enable-demuxer=rtsp --enable-indev=alsa --enable-outdev=alsa
  libavutil      57. 17.100 / 57. 17.100
  libavcodec     59. 18.100 / 59. 18.100
  libavformat    59. 16.100 / 59. 16.100
  libavdevice    59.  4.100 / 59.  4.100
  libavfilter     8. 24.100 /  8. 24.100
  libswscale      6.  4.100 /  6.  4.100
  libswresample   4.  3.100 /  4.  3.100
  libpostproc    56.  3.100 / 56.  3.100

```

**Comment by potmat** [comment]
@krehman: There's no error present in the logs you've posted.  Looks like HomeKit requests a few seconds of video, then starts another request right after (note the second "Recording request received.").  Are there more logs after this that show anything different or any errors?

Also note my statement about how HKSV works earlier in the thread:

`If you simply walk through a room, you may not see an event. This is because by the time the stream is setup and Homebridge is actually sending video to your hub(s) you've already left the room, and when your hub analyzes the video it won't see any motion, so no event will be registered.`

**Comment by potmat** [comment]
@krehman:  I've successfully gotten my second generation camera sending events.  I see the same behaviour as you, the camera detects and event and starts recording, there are no errors, however nothing shows up in the timeline.  I will keep investigating.

**Comment by krehman** [comment]
@potmat Thanks - I have been playing around with Scrypted in parallel as well, though it's a bit more stable, but my overall observation is that since Nest is cloud based, there is a delay in triggering events, especially if someone is approaching the front door. I find it's a few seconds behind. 

I have the basic Nest Aware subscription and the iCloud recording is nowhere close to what Nest provides. The live video is also not as smooth.

**Comment by potmat** [comment]
@krehman: Sorry for the delay.  I do actually see events from my second gen camera if somebody stays in the frame for a while.

Since both first and second gen cameras appear to be working, I will promote the beta version to a release version soon.

**Comment by potmat** [comment]
The beta branch has been merged to master and release 1.1.1 is out.

Please be gentle and patient about HKSV.

**Comment by potmat** [comment]
As HKSV is now "supported" I will close this issue.  Any bugs with it should require a new issue.
